src/test/integration/tests/basic.ts

```diff
@@ -344,4 +346,660 @@ export const tests: {[name: string] : SSRTest} = {
+      return html`<div class=${x}></div>`;
+    },
+    expectations: [
+      {
+        args: [nothing],
+        html: '<div class="[object Object]"></div>'
+      },
+      {
+        args: ['TEST'],
+        html: '<div class="TEST"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'AttributePart accepts a symbol': {
+    render(x: any) {
+      return html`<div class=${x}></div>`;
+    },
+    expectations: [
+      {
+        args: [Symbol()],
+        html: '<div class="Symbol()"></div>'
+      },
+      {
+        args: [Symbol()],
+        html: '<div class="Symbol()"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'AttributePart accepts a symbol with description': {
+    render(x: any) {
+      return html`<div class=${x}></div>`;
+    },
+    expectations: [
+      {
+        args: [Symbol('description!')],
+        html: '<div class="Symbol(description!)"></div>'
+      },
+      {
+        args: [Symbol('description2!')],
+        html: '<div class="Symbol(description2!)"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'AttributePart accepts an array': {
+    // TODO: Test currently fails: the default array.toString is being used
+    // during SSR, causing commas between values to be rendered. To be fixed.
+    skip: true,
+    render(x: any) {
+      return html`<div class=${x}></div>`;
+    },
+    expectations: [
+      {
+        args: [['a', 'b', 'c']],
+        html: '<div class="abc"></div>'
+      },
+      {
+        args: [['d', 'e', 'f']],
+        html: '<div class="def"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'multiple AttributeParts on same node': {
+    render(x: any, y: any) {
+      return html`<div class=${x} foo=${y}></div>`;
+    },
+    expectations: [
+      {
+        args: ['A', 'B'],
+        html: '<div class="A" foo="B"></div>'
+      },
+      {
+        args: ['C', 'D'],
+        html: '<div class="C" foo="D"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'multiple AttributeParts in same attribute': {
+    render(x: any, y: any) {
+      return html`<div class="${x} ${y}"></div>`;
+    },
+    expectations: [
+      {
+        args: ['A', 'B'],
+        html: '<div class="A B"></div>'
+      },
+      {
+        args: ['C', 'D'],
+        html: '<div class="C D"></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'multiple AttributeParts across multiple attributes': {
+    render(a: any, b: any, c: any, d: any, e: any, f: any) {
+      return html`<div ab="${a} ${b}" x c="${c}" y de="${d} ${e}" f="${f}" z></div>`;
+    },
+    expectations: [
+      {
+        args: ['a', 'b', 'c', 'd', 'e', 'f'],
+        html: '<div ab="a b" x c="c" y de="d e" f="f" z></div>'
+      },
+      {
+        args: ['A', 'B', 'C', 'D', 'E', 'F'],
+        html: '<div ab="A B" x c="C" y de="D E" f="F" z></div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
+
+  'PropertyPart accepts a string': {
```

Comment:
Are property parts treated differently for custom elements? If so, add tests for that?

Reply:
Next PR?